### PR TITLE
Adds /settings, for retrieving cook's configuration settings

### PR DESCRIPTION
**Pull request by dposada**

### CODE REVIEW

**Comment by dposada** [comment]
Do you mean use the path `/config`, or use the "literal config" instead of the parsed settings that are output from `config-settings`? If you mean the literal config, I went with the parsed settings because that seems to be what is consumed by the various parts of the system, and therefore seemed more useful for troubleshooting/inspection.

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -79,7 +82,8 @@ def test_failing_submit(self):
         self.assertEqual('failed', job['instances'][0]['status'])
 
     def test_max_runtime_exceeded(self):
-        job_uuid, resp = self.submit_job(command='sleep 60', max_runtime=5000)
+        timeout_interval_seconds = self.settings()['task-constraints']['timeout-interval-minutes'] * 60
```
Comment (shamsimam) [inline integration/tests/cook/test_basic.py]: no `get_in(self.settings() ['task-constraints' 'timeout-interval-minutes'])` available :(

Comment (shamsimam) [inline integration/tests/cook/test_basic.py]: Should we be more careful on setting this timeout, as we can end up sleeping for a really long time.

File: scheduler/src/cook/components.clj [hunk 2]
```diff
@@ -67,14 +67,15 @@
...
+  {:scheduler (fnk [mesos-datomic framework-id mesos-pending-jobs-atom [:settings task-constraints mesos-gpu-enabled is-authorized-fn :as settings]]
                 ((lazy-load-var 'cook.mesos.api/main-handler)
                   mesos-datomic
                   framework-id
                   task-constraints
                   mesos-gpu-enabled
                   (fn [] @mesos-pending-jobs-atom)
-                  is-authorized-fn))
+                  is-authorized-fn
+                  settings))
```
Comment (shamsimam) [inline scheduler/src/cook/components.clj]: We should avoid duplicating sending settings values as parameters, e.g. `task-constraints`, `mesos-gpu-enabled`, `is-authorized-fn`

File: scheduler/src/cook/mesos/api.clj [hunk 3]
```diff
@@ -1397,6 +1400,26 @@
...
                                        default-limit)))))}))
 
+;; /settings
+
+(defn stringify
+  "Converts values to strings as needed for conversion to JSON"
+  [v]
+  (cond
+    (fn? v) (str v)
+    (map? v) (map-vals stringify v)
```
Comment (shamsimam) [inline scheduler/src/cook/mesos/api.clj]: Shouldn't we also stringify the keys?

File: integration/tests/cook/test_basic.py (hunk 1 above)
Comment (dposada) [inline integration/tests/cook/test_basic.py]: re: `get_in`, I'm going to add it as a util function, coming soon
re: the timeout, `wait_for_job` has `stop_max_delay=120000` so the most we should wait is about 2 minutes

File: scheduler/src/cook/mesos/api.clj (hunk 3 above)
Comment (dposada) [inline scheduler/src/cook/mesos/api.clj]: cheshire will do that for us:

```clj
(cheshire/generate-string {:foo 1})
=> "{\"foo\":1}"
```

**Comment by dposada** [comment]
@shamsimam I addressed your feedback, thanks

**Comment by dposada** [comment]
@wyegelwel @shamsimam Any additional feedback on this?

**Comment by wyegelwel** [comment]
@dposada, your reasoning for using settings instead of config makes sense. LGTM

**Comment by dposada** [comment]
@shamsimam Any additional feedback on this?

**Pull request merged**